public/app/features/browse-dashboards/state/actions.ts

```diff
@@ -1,10 +1,89 @@
-
+import { getBackendSrv } from '@grafana/runtime';
+import { DeleteDashboardResponse } from 'app/features/manage-dashboards/types';
+import { GENERAL_FOLDER_UID } from 'app/features/search/constants';
 import { getFolderChildren } from 'app/features/search/service/folders';
+import { createAsyncThunk, DashboardDTO } from 'app/types';
+
+import { rootItemsSelector, childrenByParentUIDSelector } from './hooks';
+import { findItem } from './utils';
 
 export const fetchChildren = createAsyncThunk(
   'browseDashboards/fetchChildren',
   async (parentUID: string | undefined) => {
-    return await getFolderChildren(parentUID, undefined, true);
+    // Need to handle the case where the parentUID is the root
+    const uid = parentUID === GENERAL_FOLDER_UID ? undefined : parentUID;
+    return await getFolderChildren(uid, undefined, true);
+  }
+);
+
+export const deleteDashboard = createAsyncThunk(
+  'browseDashboards/deleteDashboard',
+  async (dashboardUID: string, thunkApi) => {
+    const result = await getBackendSrv().delete<DeleteDashboardResponse>(`/api/dashboards/uid/${dashboardUID}`);
+    const { dispatch, getState } = thunkApi;
+    const state = getState();
+
+    // refetch the parent folder children to update the tree
+    const dashboard = findItem(rootItemsSelector(state) ?? [], childrenByParentUIDSelector(state), dashboardUID);
+    dispatch(fetchChildren(dashboard?.parentUID));
+
+    return result;
+  }
+);
+
+export const deleteFolder = createAsyncThunk('browseDashboards/deleteFolder', async (folderUID: string, thunkApi) => {
+  const result = await getBackendSrv().delete(`/api/folders/${folderUID}`, undefined, {
+    params: { forceDeleteRules: true },
+  });
+  const { dispatch, getState } = thunkApi;
+  const state = getState();
+
+  // refetch the parent and destination folder children to update the tree
+  const folder = findItem(rootItemsSelector(state) ?? [], childrenByParentUIDSelector(state), folderUID);
+  dispatch(fetchChildren(folder?.parentUID));
+
+  return result;
+});
+
+export const moveDashboard = createAsyncThunk(
+  'browseDashboards/moveDashboard',
+  async ({ dashboardUID, destinationUID }: { dashboardUID: string; destinationUID: string }, thunkApi) => {
+    const fullDash: DashboardDTO = await getBackendSrv().get(`/api/dashboards/uid/${dashboardUID}`);
+    const { dispatch, getState } = thunkApi;
+    const state = getState();
+
+    const options = {
+      dashboard: fullDash.dashboard,
+      folderUid: destinationUID,
+      overwrite: false,
+    };
+
+    const result = await getBackendSrv().post('/api/dashboards/db', {
+      message: '',
+      ...options,
+    });
+
+    // refetch the parent and destination folder children to update the tree
+    const dashboard = findItem(rootItemsSelector(state) ?? [], childrenByParentUIDSelector(state), dashboardUID);
+    dispatch(fetchChildren(dashboard?.parentUID));
+    dispatch(fetchChildren(destinationUID));
+
+    return result;
```

Comment:
I wonder if we should do this (and the refetching in `deleteFolder`) in the BrowseActions view instead, _after_ we've looped over them all. This would avoid refetching affected items multiple times if we're moving multiple things...

Reply:
i'll have a quick play around and see if it looks better 👍